src/stratigraphy/metadata/coordinate_extraction.py

```diff
@@ -247,6 +263,18 @@ def _match_text_with_rect(
         lines_with_position = []
         for line in lines:
             preprocessed_text = preprocess(line.text)
+
+            # Special case for coordinates in the file: 269126143-bp.pdf. The matched string is 269578211260032 because
+            # the OCR has made a mistake between '11' and '/1'.
+            # Regex pattern to match exactly 7 digits and no '/'
+            pattern_ = r"^(?!.*\/)(?=.*\d{7})[A-Za-z0-9]*\d{7}[A-Za-z0-9]*$"
```

Comment:
I'm not happy about this change, but there was no single regex command I could find that would preserve the performance. I'm happy to spend more time trying, but I want to show you this version first, @stijnvermeeren-swisstopo.